File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/model/Result.java
package org.javamaster.invocationlab.admin.model;

import lombok.Data;
import org.apache.commons.lang3.tuple.Pair;

import java.io.Serializable;

@Data
public class Result<T> implements Serializable {
    private static final long serialVersionUID = -490238572284858235L;
    private Boolean isSuccess;
    private Integer responseCode;
    private String responseMsg;
    private T data;
    private Long count;

    public Result() {
    }

    public Result(T data) {
        this(true, 0, "请求成功", data, null);
    }

    public Result(Pair<T, Long> pair) {
        this(true, 0, "请求成功", pair.getLeft(), pair.getRight());
    }

    public Result(T data, Long count) {
        this(true, 0, "请求成功", data, count);
    }

    public Result(Integer responseCode, String responseMsg) {
        this(false, responseCode, responseMsg, null, null);

    }

    public Result(Boolean isSuccess, Integer responseCode, String responseMsg) {
        this(isSuccess, responseCode, responseMsg, null, null);
    }

    public Result(Boolean isSuccess, Integer responseCode, String responseMsg, T data, Long count) {
        this.isSuccess = isSuccess;
        this.responseCode = responseCode;
        this.responseMsg = responseMsg;
        this.data = data;
        this.count = count;
    }

    public static <T> Result<T> success(T data) {
        return new Result<>(data);
    }

    public static <T> Result<T> success(T data, long count) {
        return new Result<>(data, count);
    }

    public static <T> Result<T> fail(Integer responseCode, String responseMsg) {
        return new Result<>(responseCode, responseMsg);
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/model/Result.kt
package org.javamaster.invocationlab.admin.model

import org.javamaster.invocationlab.admin.annos.AllOpen
import org.apache.commons.lang3.tuple.Pair
import java.io.Serial
import java.io.Serializable


@AllOpen
class Result<T> : Serializable {
    var isSuccess: Boolean? = null
    var responseCode: Int? = null
    var responseMsg: String? = null
    var data: T? = null
    var count: Long? = null

    constructor()

    constructor(data: T) : this(true, 0, "请求成功", data, null)

    constructor(pair: Pair<T, Long?>) : this(true, 0, "请求成功", pair.left, pair.right)

    constructor(data: T, count: Long?) : this(true, 0, "请求成功", data, count)

    constructor(responseCode: Int?, responseMsg: String?) : this(false, responseCode, responseMsg, null, null)

    @JvmOverloads
    constructor(isSuccess: Boolean?, responseCode: Int?, responseMsg: String?, data: T? = null, count: Long? = null) {
        this.isSuccess = isSuccess
        this.responseCode = responseCode
        this.responseMsg = responseMsg
        this.data = data
        this.count = count
    }

    companion object {
        @Serial
        private val serialVersionUID = -490238572284858235L
        fun <T> success(data: T): Result<T> {
            return Result(data)
        }

        fun <T> success(data: T, count: Long): Result<T> {
            return Result(data, count)
        }

        fun <T> fail(responseCode: Int?, responseMsg: String?): Result<T> {
            return Result(responseCode, responseMsg)
        }
    }
}
